stl/src/vector_algorithms.cpp

```diff
@@ -1978,6 +1970,116 @@ namespace {
         }
     }
 
+    struct _Count_traits_8 : _Find_traits_8 {
+#ifndef _M_ARM64EC
+        static __m256i _Sub_avx(const __m256i _Lhs, const __m256i _Rhs) noexcept {
+            return _mm256_sub_epi64(_Lhs, _Rhs);
+        }
+
+        static __m128i _Sub_sse(const __m128i _Lhs, const __m128i _Rhs) noexcept {
+            return _mm_sub_epi64(_Lhs, _Rhs);
+        }
+
+        static size_t _Reduce_avx(const __m256i _Val) noexcept {
+            const __m128i _Lo64 = _mm256_extracti128_si256(_Val, 0);
+            const __m128i _Hi64 = _mm256_extracti128_si256(_Val, 1);
+            const __m128i _Rx8  = _mm_add_epi64(_Lo64, _Hi64);
+#ifdef _M_IX86
+            return _mm_cvtsi128_si32(_Rx8) + _mm_extract_epi32(_Rx8, 2);
+#else // ^^^ defined(_M_IX86) / defined(_M_X64) vvv
+            return _mm_cvtsi128_si64(_Rx8) + _mm_extract_epi64(_Rx8, 1);
+#endif // ^^^ defined(_M_X64) ^^^
+        }
+
+        static size_t _Reduce_sse(const __m128i _Val) noexcept {
+#ifdef _M_IX86
+            return _mm_cvtsi128_si32(_Val) + _mm_extract_epi32(_Val, 2);
+#else // ^^^ defined(_M_IX86) / defined(_M_X64) vvv
+            return _mm_cvtsi128_si64(_Val) + _mm_extract_epi64(_Val, 1);
+#endif // ^^^ defined(_M_X64) ^^^
+        }
+#endif // !_M_ARM64EC
+    };
+
+    struct _Count_traits_4 : _Find_traits_4 {
+#ifndef _M_ARM64EC
+        static constexpr size_t _Max_count = 0x1FFF'FFFF;

```

Comment:
This looks odd to me.

Signed vs unsigned doesn't have to do something with `0x7F...` limitation. It is rather to avoid the result overflow.

`_mm256_hadd_epi16` can operate on unsigned too, in fact, it doesn't care about signedness of input.
The problem is that adding `0x80`... values would overflow. If `_mm256_hadd_epi16` was only valid on signed types, then adding two `0x7F` would have been signed overflow, and we had to limit to `03F`.

Similarly, `_mm_cvtsi128_si32` can extract unsigned integer as well,

I'll need to recreate your test, and see why it fails.

Reply:
Recreated, pushed the changes, edited out the sign mentioning from the comments.